packages/cli/src/util/update-notifier/update-worker.js

```diff
@@ -73,3 +80,15 @@ if (process.connected && process.send) {
   console.error('No IPC bridge detected, exiting');
   process.exit(1);
 }
+
+function isRunning(lockFile) {
+  try {
+    const pid = parseInt(fs.readFileSync(lockFile, 'utf-8'));
+    process.kill(pid, 0);
+    return true;
+  } catch (e) {
```

Comment:
In this case, we don't really care if an error is thrown here. If the file doesn't exist, then no lock. If the file exists, but is empty/invalid, then treat as no lock. If the file exists, but the pid in it isn't running, then we're good to go. If the file exists and the pid is running, then we know we can bail.

Reply:
That's true, but if we need to debug an issue around here, hiding the error that gets thrown will make it harder to figure out what's going on.

I'm always skeptical of completely ignoring errors. I've encountered far too many ignored errors that would have helped me in debugging if the message was available somewhere.

Reply:
I agree that debug logging would be nice in the worker process, however that requires writing the log to a file. I'd be happy to add that if you think it's worth it.

As far as this particular function, I wouldn't be so worried about errors being thrown here. By design, it's meant to catch and return true/false.

Reply:
A log file would be nice, but I agree that it's not necessary for this. We could pass the debug flag through the IPC message and debug log it to stdio, though. I still think that would be useful.

---

The intent of the code is one thing, but things can go wrong. If I'm trying to debug an issue with the spawned worker, I would probably comment out the part where it detaches, run a CLI command with debug on, then see the log output of the worker because of the `inherit` io.

Currently, any exception thrown on these lines would be hidden. We could have a bug where writing the lockfile puts in `"234"` instead of `234`, breaking the `parseInt`, for example. We'd have to dig deep to even find the error, but if it was debug logged, we would have seen it a lot sooner in the process.